### PR TITLE
[ROCm] Support for complex type BLAS operations

### PR DESCRIPTION
**Pull request by ekuznetsov139**
This PR enables complex-type BLAS operations GEMM, GEMV, and SCAL on the ROCm platform, and  does some housekeeping (e.g. corrects misspellings) in stream_executor/rocm/rocm_blas.cc.

### CODE REVIEW

**Review by chsigg — DISMISSED** [review]
Nice!

**Comment by deven-amd** [comment]
@ekuznetsov139 , please fix the linter errors in `Ubuntu Sanity`

**Comment by gbaned** [comment]
@ekuznetsov139 Can you please resolve conflicts? Thanks!

File: tensorflow/stream_executor/rocm/rocm_blas.h [hunk 1]
```diff
@@ -121,7 +131,8 @@ class ROCMBlas : public blas::BlasSupport {
       std::unique_ptr<TemporaryDeviceMemory<
           typename RocBlasTypeConversionHelper<T>::mapped_type>> *temp_memory,
       DeviceMemory<typename RocBlasTypeConversionHelper<T>::mapped_type>
-          *device_memory);
+          *device_memory, bool copy_data,
```
Comment (chsigg) [inline tensorflow/stream_executor/rocm/rocm_blas.h]: I don't think this change belongs here.

Comment (ekuznetsov139) [reply tensorflow/stream_executor/rocm/rocm_blas.h]: 'copy_data' can be removed, but the rest of the change fixes a bug in the batched GEMM logic.

**Comment by deven-amd** [comment]
gentle ping

**Review by cheshire — APPROVED** [review]

**Review by cheshire — DISMISSED** [review]

**Review by cheshire — APPROVED** [review]

**Pull request merged**